LanguageFeatures/Augmentation-libraries/augmenting_functions_A06_t02_lib.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - An augmenting declaration uses augmented when the original declaration has
+///   no concrete implementation. Note that all external declarations are
+///   assumed to have an implementation provided by another external source, and
+///   they will throw a runtime exception when called if not.
+///
+/// @description Checks that it is not an error if an augmenting
+/// declaration uses `augmented` when the original declaration is an `external`.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_functions_A06_t02.dart';
+
+augment external void topLevelFunction() {
```

Comment:
One more thing! The grammar doesn't allow `augment external`.

I think it's supposed to be written like this:
```suggestion
augment void topLevelFunction() {
```
and it is not intended to be an error that the augmenting declaration doesn't have `external`.

In that sense, `external` plays a role which is similar to a body (so the original and the augmenting declarations can differ), not a modifier (where the original and augmenting declarations must be identical).

Reply:
Here's the error: https://github.com/dart-lang/language/blob/main/working/augmentation-libraries/feature-specification.md#:~:text=A%20declaration%20marked%20augment%20is%20also%20marked%20external.